atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepQueue.java

```diff
@@ -288,13 +307,34 @@ static SweepQueueFactory create(
                 Supplier<Integer> shardsConfig,
                 TransactionService transaction,
                 ReadBatchingRuntimeContext readBatchingRuntimeContext,
-                Function<TableReference, Optional<LogSafety>> tablesToTrackDeletions) {
+                Function<TableReference, Optional<LogSafety>> tablesToTrackDeletions,
+                SweepIndexResetProgressStage resetProgressStage) {
             Schemas.createTablesAndIndexes(TargetedSweepSchema.INSTANCE.getLatestSchema(), kvs);
+            if (resetProgressStage.shouldInvalidateOldMappings()) {
+                log.info("Invalidating old sweep mappings... now truncating sweep identifier tables.");
+
+                TargetedSweepTableFactory tableFactory = TargetedSweepTableFactory.of();
+                try {
+                    kvs.truncateTables(ImmutableSet.of(
+                            tableFactory.getSweepIdToNameTable(null).getTableRef(),
+                            tableFactory.getSweepNameToIdTable(null).getTableRef()));
+                    log.info("Successfully truncated the sweep identifier tables.");
+                } catch (Exception e) {
+                    log.warn(
+                            "A failure was observed when truncating the sweep identifier tables. If you are running"
+                                + " this as part of a broader clearance task, you MUST make sure that the success"
+                                + " message is logged BEFORE considering the reset to have been performed. Seeing this"
+                                + " message is neither an indication that the operation was success, nor is it an"
```

Comment:
nit, but doesn't really matter so ignore it given this code is going to be ripped out: was a success / was successful (former is consistent with the rest of the sentence structure)

Reply:
Perhaps worth writing in the SOP - if all nodes fail, either identify what the problem is, fix, and then bounce the nodes, or to just page someone, or some other guidance.